lavis/tasks/captioning_artist.py

```diff
@@ -0,0 +1,148 @@
+"""
+Implements evaluation for generated captions: 
+Calculate a painting's scores for mentioned artists in relation to the ground truth artists
+"""
+import logging
+from lavis.tasks.captioning import CaptionTask
+from lavis.common.registry import registry
+from torch.utils.tensorboard import SummaryWriter
+from datetime import datetime
+from lavis.common.registry import registry
+import socket
+from datetime import datetime
+import os
+from sklearn.metrics import classification_report
+from operator import itemgetter
+import json
+
+
+@registry.register_task("captioning_artist")
+class CaptionArtistTask(CaptionTask):
+    
+    def __init__(self, num_beams, max_len, min_len, evaluate, report_metric=True):
+        super().__init__(num_beams, max_len, min_len, evaluate, report_metric
+        )
+        self.writer = None
+        self.iteration = 0
+
+    # build datasets with settings from the configuration and retrieve all artists from the dataset
+    def build_datasets(self, cfg):
+        """
+        Provides adjusted implementation defined in the base_task
+        - Build pytorch dataset for the artist evaluation
+        - Extract all artists part of the dataset (necessary for score eval)
+        """
+        datasets = dict()
+        
+        datasets_config = cfg.datasets_cfg
+
+        assert len(datasets_config) > 0, "At least one dataset has to be specified."
+
+        for name in datasets_config:
+            dataset_config = datasets_config[name]
+
+            builder = registry.get_builder_class(name)(dataset_config)
+            dataset = builder.build_datasets()
+            
+            datasets[name] = dataset
+            self.artists = dataset['test'].artists
+
+            logging.info(f'The dataset consists of {len(self.artists)} artists')
+
+
+        return datasets
+
+    def setup_writer(self):
+        current_time = datetime.now().strftime("%b%d_%H-%M-%S")
+        log_dir_name =  os.path.join(registry.get_path("output_dir"),
+                "runs", current_time + socket.gethostname()
+            ) 
```

Comment:
I guess you should run an automated code formatter over all files. Usually an IDE has sth. like this